### ISSUE TITLE
[Bug]: Failing to request access for BLE permissions on device running Android 12 

### ISSUE THREAD

**Issue by tentom** [shinyorg/shiny#1066]
### Component/Nuget

BluetoothLE Client (Shiny.BluetoothLE)

### What operating system(s) are effected?

- [ ] iOS 12
- [ ] iOS 13
- [ ] iOS 14
- [ ] iOS 15
- [ ] Mac Catalyst
- [ ] Android 8
- [ ] Android 9
- [ ] Android 10
- [ ] Android 11
- [X] Android 12

### What platform(s) are effected?

- [ ] MAUI
- [ ] Xamarin Native
- [X] Xamarin Forms

### Steps To Reproduce

1. Add the new bluetooth permissions to the Android manifest (Connect and Scan permissions) 
3. Setup shiny Startup to use BLE client
```services.UseBleClient(); ```

4. Call the following in a shared project
```
var bleManager = Shiny.ShinyHost.Resolve<Shiny.BluetoothLE.IBleManager>();
var access bleManager.RequestAccess(); 
```

5. Compile with Target Android Version "Android 12.0 (API Level31 -S)" 
6. On runtime the RequestAccess() does not setup. 

### Expected Behavior

´´´access´´´ returning  as AccessState.Granted if all permissions is in the manifest and the code is executed.

### Actual Behavior

access is AccessState.NotSetup. 

Despite all the required manifest entries is in the manifest.  

Looking trough the code it seems that Shiny v2.5.5(latest release), the code on the  v2.6.0 branch and Master branches will have the same issue.

That is the code in  /[src](https://github.com/shinyorg/shiny/tree/master/src)/[Shiny.BluetoothLE](https://github.com/shinyorg/shiny/tree/master/src/Shiny.BluetoothLE)/[Platforms](https://github.com/shinyorg/shiny/tree/master/src/Shiny.BluetoothLE/Platforms)/[Android](https://github.com/shinyorg/shiny/tree/master/src/Shiny.BluetoothLE/Platforms/Android)/BleManager.cs 
Asks for the `Manifest.Permission.Bluetooth` Which is not returned from the `public static bool IsInManifest(this AndroidPlatform platform, string androidPermission)` [src](https://github.com/shinyorg/shiny/tree/728584211440ccd4d5840ea4f9c8ee330ec087ca/src)/[Shiny.Core](https://github.com/shinyorg/shiny/tree/728584211440ccd4d5840ea4f9c8ee330ec087ca/src/Shiny.Core)/[Platforms](https://github.com/shinyorg/shiny/tree/728584211440ccd4d5840ea4f9c8ee330ec087ca/src/Shiny.Core/Platforms)/[Android](https://github.com/shinyorg/shiny/tree/728584211440ccd4d5840ea4f9c8ee330ec087ca/src/Shiny.Core/Platforms/Android)/AndroidExtensions.cs

### Exception or Log output

No log output

### Code Sample

AndroidManifest.xml 
``` 
<uses-permission android:name="android.permission.BLUETOOTH" />
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
<uses-permission android:name="android.permission.BLUETOOTH_PRIVILEGED" />
<uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
<uses-permission android:name="android.permission.BLUETOOTH_CONNECT" />
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" />
```
The shinyStartup file contains
```
services.UseBleClient();
```
Somewhere in the Xamarin.forms project add these lines and make a way to trigger it.
```
var bleManager = Shiny.ShinyHost.Resolve<Shiny.BluetoothLE.IBleManager>();
AccessState accessStatus =  await bleManager.RequestAccess();
```

Executing the following code from Android native code and inspecting the requestedPermissions reveals that  `android.permission.BLUETOOTH` is not in the requestedPermissions and thus the RequestAcces code returns `NotSetup`.

```
var managerCtxt = Shiny.ShinyHost.Resolve<Shiny.BluetoothLE.Internals.ManagerContext>();
            
var permissions = managerCtxt.Android.AppContext.PackageManager?.GetPackageInfo(managerCtxt.Android.AppContext.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);

var requestedPermissions = permissions?.RequestedPermissions;

```

### Code of Conduct

- [X] My reproducible sample is NOT FROM THE SHINY SAMPLES REPOSITORY!  If you file an issue that says 'use shiny sample', YOU WILL BE BANNED FROM THE SHINY REPOS.  Read the code of conduct for why we care about this!
- [X] I am using the latest stable version from nuget (sponsors can discuss previews in discord)
- [X] I agree to follow this project's Code of Conduct

**Comment by aritchie** [comment]
Those old permissions aren't needed anymore honestly, but on my device, I am getting true for "IsInManifest".  What type of device are you running this on?

**Comment by tentom** [comment]
I'm running on a Sony Xperia 10 III (XQ-BT52).

I haven't found a way that lets me bypass the RequestAccess checks that shiny does for the BLUETOOTH manifest entry, but with bypassed for Android 12 and later I think everything will work as expected.

The problem seems to be that the old permission 
`<uses-permission android:name="android.permission.BLUETOOTH" /> `
Is not included on runtime for my device and therefore all the checks fails with notSetup. And the BLE scan is not started.

The following screenshot shows what is included on runtime from the android package manager 
![Screenshot 2022-08-09 at 09 08 33](https://user-images.githubusercontent.com/8311708/183586981-e38157d5-25cc-47c7-b6d0-2534c890b881.png)

**Comment by aritchie** [comment]
I will be updating the permissions.  As I said, those old permissions aren't needed on v12+ anyhow.  I just needed to know what I was missing because I wasn't seeing the issue.  It's a Sony device so I'm not surprised.

**Comment by tentom** [comment]
After more investigation and update to the 2.6.0 release is seems like a similar problem is occurring with the "Manifest.Permission.BluetoothAdmin"  manifest entry.

The Master branch seems to be checking for the correct manifest entries while the 2.6 branch is looking for the BluetoothAdmin permission in addition to the SCAN and CONNECT entries. 

I know the feeling of strange behaviour on devices from some Android device manufacturers, especially annoying when they break functionality that the Android team have documented should work.

**Comment by tentom** [comment]
Can confirm that this is issue is resolved with the v2.6.1 release! 

Thanks for the super fast update!
